dotcom-rendering/src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -152,27 +170,51 @@ export const SignInGateSelector = ({
 		if (gateVariant && currentTest) {
 			// eslint-disable-next-line @typescript-eslint/no-floating-promises
 			gateVariant
-				?.canShow(CAPI, !!isSignedIn, currentTest)
+				?.canShow({
+					isSignedIn: !!isSignedIn,
+					currentTest,
+					contentType,
+					sectionName,
+					tags,
+					isPaidContent,
+					isPreview,
+				})
 				.then(setCanShowGate);
 		}
-	}, [currentTest, gateVariant, CAPI, isSignedIn]);
+	}, [
+		currentTest,
+		gateVariant,
+		isSignedIn,
+		contentType,
+		sectionName,
+		tags,
+		isPaidContent,
+		isPreview,
+	]);
 
 	if (!currentTest || !gateVariant) {
 		return null;
 	}
 
-	const signInUrl = generateSignInUrl(CAPI, currentTest);
+	const signInUrl = generateSignInUrl({
+		pageId,
+		host,
+		pageViewId: window.guardian.config.ophan.pageViewId,
```

Comment:
is it worth passing this in from App.tsx as well? It's already defined there

Reply:
Good idea! I'll do this 👍